x-pack/plugins/observability_solution/investigate_app/public/pages/details/components/add_from_library_button/index.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { OverlayRef } from '@kbn/core/public';
+import { v4 } from 'uuid';
+import { openAddPanelFlyout } from '@kbn/embeddable-plugin/public';
+import { i18n } from '@kbn/i18n';
+import type { FinderAttributes } from '@kbn/saved-objects-finder-plugin/common';
+import React, { useMemo, useRef } from 'react';
+import { Item } from '@kbn/investigation-shared';
+import { CanAddNewPanel } from '@kbn/presentation-containers';
+import { EuiButtonEmpty } from '@elastic/eui';
+import { EMBEDDABLE_ITEM_TYPE } from '../../../../items/embeddable_item/register_embeddable_item';
+import { useKibana } from '../../../../hooks/use_kibana';
+interface AddFromLibraryButtonProps {
+  onItemAdd: (item: Item) => Promise<void>;
+}
+
+type InvestigationContainer = CanAddNewPanel & {
+  addNewEmbeddable: (
+    type: string,
+    explicitInput: { savedObjectId: string },
+    attributes: FinderAttributes
+  ) => Promise<{ id: string }>;
+};
+
+export function AddFromLibraryButton({ onItemAdd }: AddFromLibraryButtonProps) {
```

Comment:
nit: use `const { addItem } = useInvestigation();` so you can remove the need for the callback prop.